### ISSUE TITLE
how does search work exactly?

### ISSUE THREAD

**Issue by Ivshti** [jaruba/stremio-twitch#6]
That seems all good - https://github.com/jaruba/stremio-twitch/blob/master/twitch.js#L70

But when I search for something generic, like "the office", or even more specific, I get a ton of twitch streams in the search results (with completely un-related names). That's not normal. I can't figure out what's wrong

**Comment by jaruba** [comment]
If I search for `the office` on the twitch website, I get 99+ video results too.

It makes sense that it's search engine is made to "find whatever, even if it's an absurd result" considering that they don't have that many live streams running at the same time anyway.

The first few results do seem to have "the office" in the title of the videos on their site though, but if that doesn't happen in Stremio then it's strange.. As I can't imagine what they could be using for stream titles except the actual stream titles.

Long story short, I don't think this is something that can be "fixed". It's just a search engine made to get results no matter the cost as it searches very few items, while Stremio's search engine expects a lot more exact results as it searches a huge number of items.

The only thing we could do is filter even the twitch search results to make sure they include at least one search keyword, but we'd break more then we'd fix with that as it's complex logic and prone to error.

**Comment by Ivshti** [comment]
The thing is, often the results are videos, games or channels that really match the question.

But what we use in the add-on is LIVE results, right? For which there almost isn't anything, ever.

@jaruba

**Comment by jaruba** [comment]
Twitch gives LIVE results, yes. As the content changes constantly.

> For which there almost isn't anything, ever.

This is wrong though, if you search for game titles, Twitch will most probably give you the best content for that search query.
